hudi-client/src/main/java/org/apache/hudi/table/action/commit/WriteHelper.java

```diff
@@ -52,6 +52,10 @@
       }
       Duration indexLookupDuration = Duration.between(lookupBegin, Instant.now());
 
+      if (table.requireSortedRecords()) {
+        taggedRecords = taggedRecords.sortBy(r -> r.getRecordKey(), true, taggedRecords.getNumPartitions());
```

Comment:
This is one another shuffle which will cause performance overhead. We dont need universal order. We only need to order records that are getting written to a single partition. 
we should just do https://spark.apache.org/docs/1.2.0/api/java/org/apache/spark/rdd/OrderedRDDFunctions.html#repartitionAndSortWithinPartitions(org.apache.spark.Partitioner) when partitioning (See BaseCommitExecutor.java)

Reply:
Done. Much cleaner now.